Neos.Flow/Tests/Unit/ResourceManagement/Streams/PackageStreamWrapperTest.php

```diff
@@ -0,0 +1,124 @@
+<?php
+namespace Neos\Flow\Tests\Unit\ResourceManagement\Streams;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Package\FlowPackageInterface;
+use Neos\Flow\ResourceManagement\Exception;
+use Neos\Flow\ResourceManagement\Streams\PackageStreamWrapper;
+use org\bovigo\vfs\vfsStream;
+use Neos\Flow\Package\PackageManager;
+use Neos\Flow\Tests\UnitTestCase;
+
+/**
+ * Tests for the PackageStreamWrapper class
+ */
+class PackageStreamWrapperTest extends UnitTestCase
+{
+    /**
+     * @var PackageStreamWrapper
+     */
+    protected $packageStreamWrapper;
+
+    /**
+     * @var PackageManager|\PHPUnit\Framework\MockObject\MockObject
+     */
+    protected $mockPackageManager;
+
+    protected function setUp(): void
+    {
+        vfsStream::setup('Foo');
+
+        $this->packageStreamWrapper = new PackageStreamWrapper();
+
+        $this->mockPackageManager = $this->createMock(PackageManager::class);
+        $this->inject($this->packageStreamWrapper, 'packageManager', $this->mockPackageManager);
+    }
+
+    /**
+     * @test
+     */
+    public function openThrowsExceptionForInvalidScheme()
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        $openedPathAndFilename = '';
+        $this->packageStreamWrapper->open('invalid-scheme://foo/bar', 'r', 0, $openedPathAndFilename);
+    }
+
+    public function providePathesWithForbiddenUpwardsTraversel(): array
+    {
+        return [
+            ['package://Some.Package/../bar'],
+            ['package://Some.Package/bar/..'],
+            ['package://Some.Package/foo/../bar']
```

Comment:
this may be allowed no?

Reply:
Not sure, should it? Would probably make the adjustment in the fusionCache in the other pr obsolete

Reply:
Adjusted to be tolerant to .. at the end. Really helps to allow iteration over those such directories

Reply:
thanks ;)